static/settings.js

```diff
@@ -124,6 +126,40 @@ define(function (require) {
         });
         add(root.find('.hoverShowSource'), 'hoverShowSource', true, Checkbox);
         add(root.find('.hoverShowAsmDoc'), 'hoverShowAsmDoc', true, Checkbox);
+        var themeSelect = root.find('.theme');
+        add(themeSelect, 'theme', themes.default.id, Select,
+            _.map(themes, function (theme) {
+                return {label: theme.id, desc: theme.name};
+            })
+        );
+
+        function handleThemes() {
+            if (!colourSchemeSelect || !themeSelect) return;
```

Comment:
You're right. If this is not true, something worse than "Setting ui not found" is happening, so it does not make sense to try to _fix_ it
✔️